### DISCUSSION TITLE
AMReX builds but some tests fail

### DISCUSSION THREAD

**Discussion by tesari** [AMReX-Codes/amrex#3103] [General]
Hi,

I am using a cmake configuration file like the one listed below. On MacOS AMReX builds and all tests pass. On Red Hat Enterprise Linux Workstation release 7.9 (Maipo) AMReX build but eight tests fail:

	  2 - MultiBlock_Advection (Failed)
	  4 - Amr_Advection_AmrCore (NUMERICAL)
	  5 - Advection_AmrLevel_SV (Failed)
	  6 - Advection_AmrLevel_UV (Failed)
	 10 - LinearSolvers_ABecLap_SP (Failed)
	 11 - LinearSolvers_ABecLaplacian_C (Failed)
	 12 - LinearSolvers_NodalPoisson (Failed)
	 13 - LinearSolvers_NodeTensorLap (Failed)

For example,
2/13 Testing: MultiBlock_Advection
2/13 Test: MultiBlock_Advection
Command: "/ascldap/users/tesari/external/amrex/build_cee_gcc/Tests/MultiBlock/Advection/Test_MultiBlock_Advection"
Directory: /ascldap/users/tesari/external/amrex/build_cee_gcc/Tests/MultiBlock/Advection
"MultiBlock_Advection" start time: Jan 19 12:17 MST
Output:
----------------------------------------------------------
MPI initialized with 1 MPI processes
MPI initialized with thread support level 0
AMReX (22.04-37-g3cc4a66759c9) initialized
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
SIGABRT
See Backtrace.0 file for details
application called MPI_Abort(comm=0x84000000, 6) - process 0
[unset]: write_line error; fd=-1 buf=:cmd=abort exitcode=6
:
system msg for write_line failure : Bad file descriptor
<end of output>
Test time =  18.87 sec

What do I miss? IAMReX builds without errors or warnings, s it safe to ignore these failures? 

Thanks,
Ted

P.S. I am working on commit a5063a5d705b5192c772d5b969d13a36b97da0da (development, origin/development from Jan 17 16:22:15 2023).

cmake \
-DCMAKE_C_COMPILER=/projects/opsig/opt/gcc9.4/bin/mpicc \
-DCMAKE_CXX_COMPILER=/projects/opsig/opt/gcc9.4/bin/mpicxx \
-DCMAKE_Fortran_COMPILER=/projects/opsig/opt/gcc9.4/bin/mpif90 \
-DAMReX_PLOTFILE_TOOLS=YES \
-DAMReX_ENABLE_TESTS=YES \
-DAMReX_TESTING=YES \
-DAMReX_HDF5=YES \
-DHDF5_C_COMPILER_EXECUTABLE=/projects/opsig/opt/gcc9.4/hdf5-1.10.7/bin/h5pcc \
-DAMReX_EXPORT_DYNAMIC=YES \
-DAMReX_MPI_THREAD_MULTIPLE=YES \
-DAMReX_AMRDATA=YES \
-DAMReX_MPI=YES \
-DAMReX_PIC=YES \
-DAMReX_BUILD_SHARED_LIBS=YES \
-DCMAKE_CXX_STANDARD=17 \
-DCMAKE_INSTALL_PREFIX=/projects/opsig/opt/gcc9.4/amrex \
..

**Comment by WeiqunZhang** [comment]
I am not sure what's going on.  Could you provide more detail (maybe on a simpler test like `11 - LinearSolvers_ABecLaplacian_C (Failed)`? If there is a Backtrace.0 file, could you show it?  Is the error also `std::bad_alloc`?

**Comment by tesari** [comment]
It seems that the installation works, and I moved on.

From: Weiqun Zhang ***@***.***>
Reply-To: AMReX-Codes/amrex ***@***.***>
Date: Wednesday, January 25, 2023 at 12:06 PM
To: AMReX-Codes/amrex ***@***.***>
Cc: "Sariyski, Tzvetan Emilov" ***@***.***>, Author ***@***.***>
Subject: [EXTERNAL] Re: [AMReX-Codes/amrex] AMReX builds but some tests fail (Discussion #3103)


I am not sure what's going on. Could you provide more detail (maybe on a simpler test like 11 - LinearSolvers_ABecLaplacian_C (Failed)? If there is a Backtrace.0 file, could you show it? Is the error also std::bad_alloc?

**Comment by tesari** [comment]
Hi again,

I am installing AMReX (commit 225c605680e825e1f0793780746b4112d4262381 from Jan 2023) on another Red Hat Enterprise Linux Workstation release 7.9 (Maipo) platform. My cmake configuration script is shown below. The build with gcc is seamless - test_install and all ctests pass. The build with intel 2021 goes without errors and warnings but all tests fail. Attached are the log files from configuration, build,  and install, as well  the outputs from ‘make test_install’ and ‘cmake -V -R Parser’. Also attached are the Backtrace.0 and the output from ‘ldd amrex.so’. At this point am stuck and don’t know how to proceed. Could you advice, please?

cmake \
-DCMAKE_C_COMPILER=mpicc  \
-DCMAKE_CXX_COMPILER=mpicxx \
-DCMAKE_CXX_STANDARD=17 \
-DCMAKE_BUILD_TYPE=RelWithDebInfo \
-DAMReX_DIFFERENT_COMPILER=ON \
-DAMReX_PLOTFILE_TOOLS=YES \
-DAMReX_ENABLE_TESTS=YES \
-DAMReX_TESTING=YES \
-DAMReX_HDF5=YES \
-DHDF5_C_COMPILER_EXECUTABLE=$HDF5_INSTALL/bin/h5pcc \
-DAMReX_MPI=YES \
-DAMReX_BUILD_SHARED_LIBS=YES \
../../distributions/amrex

From: Weiqun Zhang ***@***.***>
Reply-To: AMReX-Codes/amrex ***@***.***>
Date: Wednesday, January 25, 2023 at 12:06 PM
To: AMReX-Codes/amrex ***@***.***>
Cc: "Sariyski, Tzvetan Emilov" ***@***.***>, Author ***@***.***>
Subject: [EXTERNAL] Re: [AMReX-Codes/amrex] AMReX builds but some tests fail (Discussion #3103)


I am not sure what's going on. Could you provide more detail (maybe on a simpler test like 11 - LinearSolvers_ABecLaplacian_C (Failed)? If there is a Backtrace.0 file, could you show it? Is the error also std::bad_alloc?

**Reply by WeiqunZhang** [reply]
Don't see any logs.  For attachment, you may have to do it on the github website, or you forgot to attach them in email.

**Reply by tesari** [reply]
Strange, because on the bcc to myself I got the attachments. I attached them now from the website.

**Comment by tesari** [comment]
Backtrace.0
cmake.log
install.log
ldd_amrex.so.log
make.log
Parser.log
test_install.log

**Comment by tesari** [comment]
I sent the attachments through the github website. Please let me know if you have not got them.

From: Weiqun Zhang ***@***.***>
Reply-To: AMReX-Codes/amrex ***@***.***>
Date: Monday, January 30, 2023 at 2:24 PM
To: AMReX-Codes/amrex ***@***.***>
Cc: "Sariyski, Tzvetan Emilov" ***@***.***>, Author ***@***.***>
Subject: [EXTERNAL] Re: [AMReX-Codes/amrex] AMReX builds but some tests fail (Discussion #3103)


Don't see any logs. For attachment, you may have to do it on the github website, or you forgot to attach them in email.

**Reply by WeiqunZhang** [reply]
Still no.  I also asked @atmyers.  He doesn't see any attachments either.

**Reply by WeiqunZhang** [reply]
You can log out of your github account to see if you can still see it.

**Comment by tesari** [comment]
I did not know that github does not support file types log and txt. I renamed make.log to make_log, etc.

From: Weiqun Zhang ***@***.***>
Reply-To: AMReX-Codes/amrex ***@***.***>
Date: Monday, January 30, 2023 at 3:42 PM
To: AMReX-Codes/amrex ***@***.***>
Cc: "Sariyski, Tzvetan Emilov" ***@***.***>, Author ***@***.***>
Subject: [EXTERNAL] Re: [AMReX-Codes/amrex] AMReX builds but some tests fail (Discussion #3103)


You can log out of your github account to see if you can still see it.

**Comment by tesari** [comment]
Sorry for the confusion.

[Backtrace.txt](https://github.com/AMReX-Codes/amrex/files/10540685/Backtrace.txt)
[cmake.txt](https://github.com/AMReX-Codes/amrex/files/10540695/cmake.txt)
[install.txt](https://github.com/AMReX-Codes/amrex/files/10540699/install.txt)
[ldd_amrex_so.txt](https://github.com/AMReX-Codes/amrex/files/10540701/ldd_amrex_so.txt)
[make.txt](https://github.com/AMReX-Codes/amrex/files/10540702/make.txt)
[Parser.txt](https://github.com/AMReX-Codes/amrex/files/10540703/Parser.txt)
[test_install.txt](https://github.com/AMReX-Codes/amrex/files/10540706/test_install.txt)

**Comment by tesari** [comment]
I uploaded the missing attachments. Please let me know if you need more details … and sorry for the confusion.

From: Weiqun Zhang ***@***.***>
Reply-To: AMReX-Codes/amrex ***@***.***>
Date: Monday, January 30, 2023 at 3:42 PM
To: AMReX-Codes/amrex ***@***.***>
Cc: "Sariyski, Tzvetan Emilov" ***@***.***>, Author ***@***.***>
Subject: [EXTERNAL] Re: [AMReX-Codes/amrex] AMReX builds but some tests fail (Discussion #3103)


You can log out of your github account to see if you can still see it.
